File: src/main/kotlin/dev/jefy/connectpro/engagement/applicaion/command/ReviewCommand.java
package dev.jefy.connectpro.engagement.applicaion.command;

import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.engagement.applicaion.dtos.ReviewRequest;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface ReviewCommand {
    void createOrUpdate(ServiceId serviceId, ReviewRequest request);
    void delete(ServiceId serviceId);
}

File: src/main/kotlin/dev/jefy/connectpro/engagement/application/command/ReviewCommand.kt
package dev.jefy.connectpro.engagement.application.command

import dev.jefy.connectpro.engagement.application.dtos.ReviewRequest
import dev.jefy.connectpro.engagement.presentation.ReviewOperationResult
import dev.jefy.connectpro.portfolio.domain.vo.ServiceId

/**
 * @author Jôph Yamba
 */
interface ReviewCommand {
    fun createOrUpdate(serviceId: ServiceId, request: ReviewRequest): ReviewOperationResult
    fun delete(serviceId: ServiceId)
}
